tests/qchem/test_factorization.py

```diff
@@ -167,3 +167,41 @@ def test_empty_error(two_tensor):
 
     with pytest.raises(ValueError, match="All eigenvectors are discarded."):
         qml.qchem.factorize(two_tensor, 1e-5, 1e1)
+
+
+@pytest.mark.parametrize(
+    ("one_matrix", "two_tensor", "tol_factor", "coeffs_ref"),
+    [
+        (  # one_matrix and two_tensor are obtained with:
+            # symbols  = ['H', 'H']
+            # geometry = np.array([[0.0, 0.0, 0.0], [1.39839789, 0.0, 0.0]], requires_grad = False)
+            # mol = qml.qchem.Molecule(symbols, geometry)
+            # core, one_matrix, two_tensor = qml.qchem.electron_integrals(mol)()
+            np.array([[-1.25330961e00, -2.07722728e-13], [-2.07611706e-13, -4.75069041e-01]]),
+            np.array(  # two-electron integral tensor in physicist notation
+                [
+                    [
+                        [[6.74755872e-01, 2.39697151e-13], [2.39780418e-13, 1.81210478e-01]],
+                        [[2.39808173e-13, 1.81210478e-01], [6.63711349e-01, 2.21378471e-13]],
+                    ],
+                    [
+                        [[2.39808173e-13, 6.63711349e-01], [1.81210478e-01, 2.21822560e-13]],
+                        [[1.81210478e-01, 2.21489493e-13], [2.21267449e-13, 6.97651447e-01]],
+                    ],
+                ]
+            ),
+            1.0e-5,
+            [
+                np.array([-1.29789639, 0.84064639, 0.45725]),
+                np.array([-4.86900854e-05, -2.75009253e-03, 5.59756523e-03, -2.79878262e-03]),
+                np.array([0.09060524, -0.04530262, -0.04530262]),
+                np.array([-0.34038856, 0.50911518, -0.33745324, 0.16872662]),
+            ],
+        ),
+    ],
+)
+def test_basis_rotation(one_matrix, two_tensor, tol_factor, coeffs_ref):
```

Comment:
I can't think of any more tests at the top of my head, but this seems insufficient for a thorough coverage of this function. I encourage you to spend more time thinking about additional tests we can perform